test/hotspot/jtreg/compiler/c2/TestMergeStores.java

```diff
@@ -54,6 +54,8 @@ public class TestMergeStores {
     private static final Unsafe UNSAFE = Unsafe.getUnsafe();
     private static final Random RANDOM = Utils.getRandomInstance();
 
+    private static final boolean IS_BIG_ENDIAN = UNSAFE.isBigEndian();
```

Comment:
`static` is very important here, so that the `if` constant fold in the test. Otherwise we don't know if we have the IR rule pass because of the correct branch. Maybe add a comment for that.

Reply:
Sure. I assumed that is clear to people looking at jit compiler tests :)
I removed `IS_BIG_ENDIAN` again since it wasn't needed anymore with the last comit (3169a3104b7323c4ff6f2714449a7c28025d0bba).